account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1168 @@
+        # 'chatter_msg': ['Notes added in chatter of the invoice'],
+        # 'note': 'Note embedded in the document',
+        # 'origin': 'Origin note',
+        # 'lines': [{
+        #       'product': {
+        #           'barcode': '4123456000021',
+        #           'code': 'GZ250',
+        #           },
+        #       'name': 'Gelierzucker Extra 250g',
+        #       'price_unit': 1.45, # price_unit without taxes
+        #       'qty': 2.0,
+        #       'price_subtotal': 2.90,  # not required, but needed
+        #               to be able to generate adjustment lines when decimal
+        #               precision is not high enough in Odoo
+        #       'uom': {'unece_code': 'C62'},
+        #       'taxes': [{
+        #           'amount_type': 'percent',
+        #           'amount': 20.0,
+        #           'unece_type_code': 'VAT',
+        #           'unece_categ_code': 'S',
+        #           'unece_due_date_code': '432',
+        #           }],
+        #       'date_start': '2015-10-01',
+        #       'date_end': '2015-10-31',
+        #       # date_start and date_end on lines override the global value
+        #       }],
+        # }
+
+        # IMPORT CONFIG
+        # {
+        # 'invoice_line_method': '1line_no_product',
+        # 'account_analytic': Analytic account recordset,
+        # 'account': Account recordset,
+        # 'taxes': taxes multi-recordset,
+        # 'label': 'Force invoice line description',
+        # 'product': product recordset,
+        # }
+        #
+        # Note: we also support importing customer invoices via
+        # create_invoice() but only with 'nline_*' invoice import methods.
+
+    @api.model
+    def _prepare_create_invoice_vals(self, parsed_inv, import_config=False):
+        assert parsed_inv.get("pre-processed"), "pre-processing not done"
+        # WARNING: on future versions, import_config will probably become
+        # a required argument
+        aio = self.env["account.move"]
+        ailo = self.env["account.move.line"]
+        bdio = self.env["business.document.import"]
+        rpo = self.env["res.partner"]
+        company = (
+            self.env["res.company"].browse(self.env.context.get("force_company"))
+            or self.env.user.company_id
+        )
+        start_end_dates_installed = hasattr(ailo, "start_date") and hasattr(
+            ailo, "end_date"
+        )
+        if parsed_inv["type"] in ("out_invoice", "out_refund"):
+            partner_type = "customer"
+        else:
+            partner_type = "supplier"
+        partner = bdio._match_partner(
+            parsed_inv["partner"], parsed_inv["chatter_msg"], partner_type=partner_type
+        )
+        partner = partner.commercial_partner_id
+        currency = bdio._match_currency(
+            parsed_inv.get("currency"), parsed_inv["chatter_msg"]
+        )
+        journal_id = (
+            aio.with_context(type=parsed_inv["type"], company_id=company.id)
```

Comment:
Should be default_type

Reply:
I think this comment still applies here: https://github.com/OCA/edi/pull/213/files#diff-4e91a94c7b31b4f68cb4631aafdd179e6a84e426b1d5d955b1650f68b3d7cb70R71